### ISSUE TITLE
Bump upper bounds to work with stackage-nightly (ghc 8.2)

### ISSUE THREAD

**Issue by Drezil** [thoughtbot/yesod-auth-oauth2#89]
WARNING: Ignoring out of range dependency (allow-newer enabled): aeson-1.2.3.0. yesod-auth-oauth2 requires: >=0.6 && <1.2
WARNING: Ignoring out of range dependency (allow-newer enabled): hoauth2-1.5.0. yesod-auth-oauth2 requires: >=1.3.0 && <1.4

i had to force "allow-newer". No compile-errors though. Looks like the bump for those two packages would be no problem.
(at least for the current stackage-nightly that will become the next lts-10.x for ghc 8.2)

**Comment by pbrisbin** [comment]
Thanks for the report. I wanted to reproduce this before fixing it, but I'm having some trouble:

```
% stack build --resolver nightly --pedantic --test
Selected resolver: nightly-2017-12-10
Downloaded nightly-2017-12-10 build plan.    
AesonException "Error in $.packages.cassava.constraints.flags['bytestring--lt-0_10_4']: Invalid flag name: \"bytestring--lt-0_10_4\""
```

Do you know anything about this?

Anyway, I'm going to try and make a branch that checks compilation with `nightly` on CI; maybe using stack-1.6.1 (which I can't do yet locally) resolves that issue. Then I can see the failure, confirm the fix, and catch these (a little) more pro-actively going forward.

**Comment by Drezil** [comment]
stack 1.5.x breaks with ghc8.2, due to ghc forcing things internally. It worked until dec 6th with 1.5.1 iirc. then ghc broke things (that were already fixed in stack 1.6). But stack 1.6 released 3 days later .. so that should resolve many things.

Some minor things have to be changed in the cabal-file to pass testing with -Wall. PR is on the way.

Log for `stack build --resolver nightly --pedantic --test` with resolver `nightly-2017-12-13`:
```
Progress: 1/2
URI.ByteString.Extension
  IsString Scheme
    works
  IsString Host
    works
  IsString URIRef Relative
    works
  IsString URIRef Absolute
    works
  fromText
    returns Just a URI for valid values, as the quasi-quoter would
    returns Nothing for invalid values
  unsafeFromText
    returns a URI for valid values, as the quasi-quoter would
    raises for invalid values
  toText
    serializes the URI to text
  fromRelative
    makes a URI absolute with a given host
  withQuery
    appends a query to a URI
    handles a URI with an existing query
    handles santization of the query

Finished in 0.0009 seconds
13 examples, 0 failures
             
yesod-auth-oauth2-0.3.0: Test suite test passed
```

**Comment by pbrisbin** [comment]
Fixed with #90.
